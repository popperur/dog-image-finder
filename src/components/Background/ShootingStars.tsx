import {
  ShootingStar,
  ShootingStarContainer,
} from "components/Background/styles.tsx";
import useAnimationTimer from "hooks/useAnimationTimer.ts";

function ShootingStars() {
  const [keyIndex, animationDuration] = useAnimationTimer({
    initial: { duration: 4, delay: 6 },
    durationFn: () => 4 + Math.random() * 4,
    delayFn: () => 6 + Math.random() * 14,
  });

  return (
    <ShootingStarContainer>
      <ShootingStar
        $duration={animationDuration}
        key={`shooting-star-${keyIndex}`}
      />
    </ShootingStarContainer>
  );
}

export default ShootingStars;
