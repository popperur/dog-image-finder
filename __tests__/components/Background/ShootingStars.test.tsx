import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import ShootingStars from "components/Background/ShootingStars.tsx";

describe("ShootingStars component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<ShootingStars />);

    expect(asFragment()).toMatchSnapshot();
  });
});
