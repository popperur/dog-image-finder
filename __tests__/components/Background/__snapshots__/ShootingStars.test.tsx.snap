// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShootingStars component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  top: 0;
  height: 51%;
  width: 55%;
}

.c1 {
  position: relative;
  top: -20%;
  left: -20%;
  height: 80px;
  width: 2px;
  background-color: #fff;
  box-shadow: 0 0 30px #fff;
  transform-origin: 10% 0;
  transform: rotate(130deg) scale(0);
  animation: giQQSW ease-in;
  animation-duration: 4s;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;
